blueocean-core-js/src/js/utils/jenkins-encode.js

```diff
@@ -0,0 +1,31 @@
+export default class JenkinsEncode {
+    static _uriMap = null;
+    static _getUrlMap() {
+        if(this._uriMap == null) {
+            const raw = "!  $ &'()*+,-. 0123456789   =  @ABCDEFGHIJKLMNOPQRSTUVWXYZ    _ abcdefghijklmnopqrstuvwxyz";
```

Comment:
:bug: this stuff should just be a static property somewhere

Reply:
why so many spaces?

Reply:
Spaces are the character placeholders where we want to encode lol

Reply:
oooh. funny